lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -1246,7 +1344,7 @@ namespace {
       // a 'default' to be inserted.
       // FIXME: Do something sensible for non-frozen enums.
       if (!sawRedundantPattern &&
-          coveredSpace.getSize(TC, DC) >= totalSpaceSize)
+          coveredSpace.getSize(TC, DC) >= totalSpace.getSize(TC, DC))
```

Comment:
Can you look into completely removing this clause?  It is not correct in general, and should be subsumed by your check if it isn't already.

Reply:
I do suggest doing that separately, so we can revert it separately if need be.

Reply:
I'll look into doing that separately